File: com/supcon/common/view/base/controller/BaseController.java
package com.supcon.common.view.base.controller;

import android.content.Intent;

import com.supcon.common.view.Lifecycle;
import com.supcon.common.view.base.IData;

import java.util.Map;

/**
 * Created by wangshizhan on 17/6/29.
 */

public  class BaseController implements Lifecycle, IData{

    /**
     * 最初初始化
     */
    public void onInit(){

    }

    /**
     * 初始化VIEW
     */
    public void initView() {

    }

    /**
     * 初始化事件监听
     */
    public void initListener() {

    }

    /**
     * 初始化数据
     */
    public void initData() {

    }

    /**
     * activity 或 fragment onStart
     */
    public void onStart() {

    }

    /**
     * activity 或 fragment onStop
     */
    public void onStop() {

    }

    /**
     * activity 或 fragment onResume
     */
    public void onResume() {

    }

    /**
     * activity 或 fragment onPause
     */
    public void onPause() {
    }

    /**
     * activity 或 fragment onDestroy
     */
    public void onDestroy() {

    }

    /**
     * activity 或 fragment onRetry
     */
    public void onRetry() {

    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {

    }

    @Override
    public void refresh() {

    }

    @Override
    public boolean checkBeforeSubmit(Map<String, Object> map) {
        return doSave(map);
    }

    @Override
    public boolean doSave(Map<String, Object> map) {
        return true;
    }

    @Override
    public boolean isModified() {
        return false;
    }
}

File: module_middleware/src/main/java/com/yaobing/module_middleware/BaseController.kt
package com.yaobing.module_middleware

import android.content.Intent

/**
 * @author : yaobing
 * @date   : 2022/08/01 14:46
 * @desc   : 要求所有子类controller都要继承baseController的生命周期，便于管控，防止内存泄漏之类的
 */
class BaseController : Lifecycle,IData {


    //为controller绑定的context所在。比如activity/fragment
    private var target: Any? = null

    fun setTarget(target: Any?) {
        this.target = target
    }

    fun getTarget(): Any? {
        return target
    }

    fun <T> getActivity(target: T): T {
        return target!!::class.java.cast(target)!!
    }
    //------------------------------------------------下面的都是重写的生命周期的方法

    override fun refresh() {
        
    }

    override fun checkBeforeSubmit(map: Map<String?, Any?>?): Boolean {
        return doSave(map)
    }

    override fun doSave(map: Map<String?, Any?>?): Boolean {
        return true
    }

    override fun isModified(): Boolean {
        return false
    }

    override fun onInit() {
        
    }

    override fun initView() {
        
    }

    override fun initListener() {
        
    }

    override fun initData() {
        
    }

    override fun onStart() {
        
    }

    override fun onStop() {
        
    }

    override fun onResume() {
        
    }

    override fun onPause() {
        
    }

    override fun onDestroy() {
        
    }

    override fun onRetry() {
        
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        
    }
}